src/main/java/com/adobe/target/edge/client/service/DefaultTargetService.java

```diff
@@ -56,23 +62,55 @@ public DefaultTargetService(ClientConfig clientConfig) {
     this.targetHttpClient.addDefaultHeader(SDK_USER_KEY, SDK_USER_VALUE);
     this.targetHttpClient.addDefaultHeader(SDK_VERSION_KEY, SDK_VERSION);
     this.clientConfig = clientConfig;
+    this.telemetryService = new TelemetryService(clientConfig);
   }
 
   @Override
   public TargetDeliveryResponse executeRequest(TargetDeliveryRequest deliveryRequest) {
+    TimingTool timer = new TimingTool();
+    timer.timeStart(TIMING_EXECUTE_REQUEST);
+
     String url = clientConfig.getUrl(getBestLocationHint(deliveryRequest));
+    TargetDeliveryResponse targetDeliveryResponse;
+    synchronized (this) {
+      if(!storedTelemetries.isEmpty()) {
+        Telemetry telemetry = new Telemetry();
+        deliveryRequest.getDeliveryRequest().setTelemetry(telemetry);
```

Comment:
As suggested above, let's move these details to `TelemetryService`, so we only have something like this:
```suggestion
        Telemetry telemetry = telemetryService.getTelemetry();
        if (!telemetry.entries.isEmpty()) {
          deliveryRequest.getDeliveryRequest().setTelemetry(telemetry);
        }
```